src/geography.cpp

```diff
@@ -133,26 +137,78 @@ PyObjectGeography destroy_prepared(PyObjectGeography obj) {
 void init_geography(py::module &m) {
     // Geography types
 
-    py::enum_<GeographyType>(m, "GeographyType")
-        .value("NONE", GeographyType::None)
-        .value("POINT", GeographyType::Point)
-        .value("LINESTRING", GeographyType::LineString);
+    auto pygeography_types =
+        py::enum_<GeographyType>(m, "GeographyType", R"pbdoc(
+        The enumeration of Geography types
+    )pbdoc");
+
+    pygeography_types.value("NONE", GeographyType::None);
+    pygeography_types.value("POINT", GeographyType::Point);
+    pygeography_types.value("LINESTRING", GeographyType::LineString);
 
     // Geography classes
 
-    py::class_<Geography>(m, "Geography")
-        .def_property_readonly("dimensions", &Geography::dimension)
-        .def_property_readonly("nshape", &Geography::num_shapes)
-        .def("__repr__", [](const Geography &geog) {
-            s2geog::WKTWriter writer;
-            return writer.write_feature(geog.geog());
-        });
+    auto pygeography = py::class_<Geography>(m, "Geography", R"pbdoc(
+        Base class for all geography types.
+
+        Cannot be instanciated directly.
+
+    )pbdoc");
+
+    pygeography.def_property_readonly("dimensions", &Geography::dimension,
+                                      R"pbdoc(
+        Returns the inherent dimensionality of a geometry.
+
+        The inherent dimension is 0 for points, 1 for linestrings and 2 for
+        polygons. For geometrycollections it is the max of the containing elements.
+        Empty collections and None values return -1.
+
+    )pbdoc");
+
+    pygeography.def_property_readonly("nshape", &Geography::num_shapes, R"pbdoc(
```

Comment:
(not directly related to this PR, just thinking about it while looking through this)

We could also keep the `num_shapes` naming on the python side? (to keep it consistent with c++, and I also find it a bit more readable, and both are different from shapely anyway)

Reply:
Agreed.